src/app/navbar/navbar.component.html

```diff
@@ -41,10 +41,16 @@
           <img class="site-icons-small hidden-sm" src="../assets/svg/main-nav/icons-navigation-forum.svg" alt="stacked pages" /> Forum
         </a>
         <span class="spacer"></span>
-        <a *ngIf="!isLoggedIn" routerLink="/login">
-          <button mat-raised-button color="accent">Login</button>
-        </a>
-        <template [ngTemplateOutlet]="loginButton"></template>
+        <div fxLayoutGap="1rem">
+          <a *ngIf="!isLoggedIn" routerLink="/register" [state]="{ data: 1 }">
```

Comment:
Why do you need to pass `state` here and on line 49? Can't you just go off like the url, like you are already doing, in login.component.ts?

Reply:
Thank you, just tried it and I don't need to pass state. I think I initially had this because I wasn't going to make a new URL for register.